### DISCUSSION TITLE
How can a non-root user validate the nerdctl installed by root user is trusted?

### DISCUSSION THREAD

**Discussion by zibinpan** [containerd/nerdctl#2925] [Q&A]
In the scenario that the root user cannot be trusted, how can a non-root user validate the nerdctl installed by root user is trusted? Otherwise, it's in danger when the non-root user building its container.

**Comment by AkihiroSuda** [comment]
> In the scenario that the root user cannot be trusted

In this scenario no program can be trusted.

**Reply by zibinpan** [reply]
Can the non-root user scan the file of nerdctl to validate its consistency? Of course, the root user is required to open necessary authority for non-root user to do this scanning.

**Reply by AkihiroSuda** [reply]
No, because the root user may intercept the syscalls etc. to fake up the scanning result

**Reply by zibinpan** [reply]
Thanks. But I am very confusing that if I want the rooter to build a TEE (Trusted execution environment) container, how can it be trusted? Or there doesn't exist any totally trusted TEE technique?
